### ISSUE TITLE
Page Items to Submit: Binds that do not start with PX_ are not added

### ISSUE THREAD

**Issue by martindsouza** [foex-open-source/apex-builder-extension-by-fos#23]
On `P1` I have a classic report that looks like this:

```sql
select e.*
from emp e
where 1=1
  and e.deptno = :P1_DEPT_NO
  and e.name like :NAME
```

After saving the report the `Page Items to Submit` are automatically updated and set to `P1_DEPT_NO`. Note that `NAME` is a page item on my page (have our reasons for this) that is not prefixed with `P1_`. I then have to manually add in `,NAME` to the `Page Items to Submit`. The issue is that each time I make a change to the report I have to re-add in `,NAME` to `Page Items to Submit` which we may not remember to do each time.

**Comment by foex-open-source** [comment]
Thanks for making us aware of this bug, we'll look into it.
In the next release - of course, with a fix for this issue - the feature will be optional ( ie. can be turned off ), so these situations can be handled / avoided.

**Comment by martindsouza** [comment]
@foex-open-source not sure what you mean that the feature will be optional. I was going to suggest that all binds are captured but understand why you don’t want it to (ex: app items). Perhaps the rules should be this:

- Get list of binds used in query
- If list has binds that aren’t in query, remove
- Any page items added to list only if they don’t exist

Thoughts?

**Comment by foex-open-source** [comment]
Yes, the rules look good. Actually, this is how it works more-or-less at the moment too, with the difference that we assume conventional page item names (P+pageId+_+....). This will be changed to look for every bind variable.

"Optional" means that a "toggle" menu entry will be added to the "Settings" menu and with that the feature can be disabled/enabled immediately, so if an annoying bug/unexpected behaviour appears, you can just turn the feature off, no need to disable the whole extension.

<img width="325" alt="image" src="https://user-images.githubusercontent.com/61969891/167457364-ba47bd3e-eb3b-45f2-81e5-13c0d54325ee.png">

What do you think?

**Comment by martindsouza** [comment]
Yes looks good with me. Thanks
